New: CSS view transitions module landing page (#34803)
mdn/content: files/en-us/web/css/css_view_transitions/index.md

---
title: CSS view transitions
slug: Web/CSS/CSS_view_transitions
page-type: css-module
spec-urls:
  - https://drafts.csswg.org/css-view-transitions-1/
  - https://drafts.csswg.org/css-view-transitions-2/
---

{{CSSRef}}

> The **CSS view transitions** module defines the [View Transition API](/en-US/docs/Web/API/View_Transitions_API), along with associated properties and pseudo-elements, which allows developers to create animated visual transitions when the state of a document changes.

Author: updated

## Reference

### Properties

> - {{cssxref("view-transition-name")}}

Reviewer: should be marked as experimental?

Author: updated

Reviewer: thanks

> [!NOTE]

> > The CSS view transitions module also defines an additional property, `view-transition-class`, which is not currently supported in any browser.

Reviewer: ```suggestion
> The module defines the CSS property `view-transition-class`, but it is currently not supported in any browser.
```

### At-rules and descriptors

- {{cssxref("@view-transition")}}

>   - [`navigation`](/en-US/docs/Web/CSS/@view-transition#navigation) descriptor

Reviewer: Maybe we can drop the descriptor since this at-rule has just this one? The heading in that case can be just "At-rules"

Author: keeping as we want to list everything

> ### Selectors

Reviewer: Pseudo-elements?

Author: Good catch. changed to selectors and pseudo elements

- {{cssxref("::view-transition")}}
- {{cssxref("::view-transition-image-pair()")}}
- {{cssxref("::view-transition-group()")}}
- {{cssxref("::view-transition-new()")}}

> - {{cssxref("::view-transition-old()")}}

Reviewer: all experimental

Author: done

> [!NOTE]

> > The CSS view transitions module also defines two additional pseudo-class, `:active-view-transition` and the functional pseudo-classs `:active-view-transition-type()`, which are not yet supported in any browser.

Reviewer: ```suggestion
> The module defines the pseudo-classes `:active-view-transition` and `:active-view-transition-type()`, but they are currently not supported in any browser.
```

Author: went with 

> This module also defines the `:active-view-transition` and `:active-view-transition-type()`  pseudo-classes, but they are currently not supported in any browser.

### Interfaces

- {{domxref("CSSViewTransitionRule")}}
- {{domxref("ViewTransition")}}
  - {{domxref("ViewTransition.skipTransition()")}} method
  - {{domxref("ViewTransition.updateCallbackDone")}}
  - {{domxref("ViewTransition.ready")}}
  - {{domxref("ViewTransition.finished")}}
- {{domxref("Document.startViewTransition()")}} method

## Guides

- [Using the View Transitions API](/en-US/docs/Web/API/View_Transitions_API/Using)

>   - : Create view transitions and customize transition animations, including manipulating active view transitions.

Reviewer: ```suggestion
  - : Explains how to create view transitions and customize transition animations, including manipulating active view transitions.
```

Author: updated

## Related concepts

- {{domxref("PageRevealEvent", "pagereveal")}} event
- {{domxref("PageSwapEvent", "pageswap")}} event
- {{domxref("Document.visibilityState")}}

> [CSS animations](/en-US/docs/Web/CSS/CSS_animations) module
> 
> - {{cssxref("animation")}}
> - {{cssxref("@keyframes")}}
> - {{domxref("CSSKeyframesRule")}}
> - {{domxref("CSSStyleRule")}}
> - [Web animations API](/en-US/docs/Web/API/Web_Animations_API)

Reviewer: ```suggestion
- [CSS animations](/en-US/docs/Web/CSS/CSS_animations) module:

  - {{cssxref("animation")}} property
  - {{cssxref("@keyframes")}} at-rule
  - {{domxref("CSSKeyframesRule")}} interface
  - {{domxref("CSSStyleRule")}} interface
  - [Web animations API](/en-US/docs/Web/API/Web_Animations_API)
```

Author: updated

[CSS transforms](/en-US/docs/Web/CSS/CSS_transforms) module

- {{cssxref("transform")}}

> - {{cssxref("transform-function")}}

Reviewer: similar indentation, colon, and adjectives for the items in the list

Author: done
